### ISSUE TITLE
Add support for Auth0

### ISSUE THREAD

**Issue by mraible** [jhipster/generator-jhipster#14864]
##### **Overview of the feature request**

With [Okta and Auth0 joining forces today](https://www.okta.com/blog/2021/04/its-official-okta-joins-forces-with-auth0/), it'd be cool to document how to integrate JHipster with Auth0. Even better, maybe we can add JHipster support to the [Auth0 CLI](https://auth0.com/blog/introducing-auth0-cli/)!

I'm not sure we need to do much on the JHipster side, but I did discover some issues to resolve.

- Figure out how to add a `groups` claim. [Add user roles to tokens](https://auth0.com/docs/authorization/sample-use-cases-rules-with-authorization#add-user-roles-to-tokens) shows you how to do it, but it seems a namespace prefix might be required.
- Logout fails because there’s no `end_session_endpoint` in `.well-known/openid-configuration`. 
- Figure out how to put the email in the `sub` claim so the welcome message is friendlier. 

<img width="937" alt="Screen Shot 2021-05-03 at 2 09 14 PM" src="https://user-images.githubusercontent.com/17892/116931113-d99e6e80-ac1d-11eb-9107-eb4977083912.png">

##### **Motivation for or Use Case**

More IdP options for developers seems like a good thing.

##### **Related issues or PR**

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by mraible** [comment]
Thanks for looking into this @Falydoor. Auth0's [Spring Boot example](https://auth0.com/docs/quickstart/webapp/java-spring-boot/01-login#add-logout-to-your-application) has a hardcoded logout URI too.

**Comment by Falydoor** [comment]
How do you want to proceed @mraible ?

I can fix the login/logout but could not have the groups working, can you elaborate on that?

**Comment by mraible** [comment]
@Falydoor I can talk to some of the Auth0 folks and see if I can get a resolution on these issues.

**Comment by mraible** [comment]
Please keep it open. I might not be able to look at before July though. I'm on vacation for July, returning in August.

**Comment by vishal423** [comment]
I have a working prototype with `Auth0`, will raise a PR in a day or two

**Comment by mraible** [comment]
@vishal423 I tried the integration today using JHipster 7.3.0. I used a JHipster tutorial I wrote a couple of weeks ago. It shows the user is logged in, but the navbar doesn't reflect it when using React.

<img width="1712" alt="Screen Shot 2021-10-12 at 1 36 28 AM" src="https://user-images.githubusercontent.com/17892/136913099-278f23ae-5083-4466-94e1-b1d2c097be4a.png">

My blog post's PR is at https://github.com/oktadev/okta-blog/pull/891. It should have a Netlify preview link to see the rendered post.
